File: Easy/2026-05-29-3300-Minimum-Element-After-Replacement-With-Digit-Sum/solution.js
var minimumElement = function(nums) {
    const digitSum = (num) => {
        let sum = 0;
        while (num > 0) {
            sum += num % 10;
            num = Math.floor(num / 10);
        }
        return sum;
    };

    let minElement = Infinity;
    for (let num of nums) {
        minElement = Math.min(minElement, digitSum(num));
    }
    
    return minElement;
};

File: Easy/2026-05-29-3300-Minimum-Element-After-Replacement-With-Digit-Sum/solution.java
class Solution {
    public int minimumElementAfterReplacement(int[] nums) {
        int minElement = Integer.MAX_VALUE;
        
        for (int num : nums) {
            minElement = Math.min(minElement, digitSum(num));
        }
        
        return minElement;
    }
    
    private int digitSum(int num) {
        int sum = 0;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
}